src/SOS/lldbplugin/services.cpp

```diff
@@ -757,15 +757,30 @@ LLDBServices::GetContextStackTrace(
 // IDebugDataSpaces
 //----------------------------------------------------------------------------
 
+#ifndef PAGE_SIZE 
+#define PAGE_SIZE 0x1000
```

Comment:
Should this be const int64_t since offset is ULONG64

Reply:
This will likely not be true for macOS - hence the sysconf call in the runtime.

Reply:
I can't see anything in older lldbs. There's SBMemoryRegion::GetPageSize for newer versions, but not sure that'd work.

Reply:
I couldn't find a lldb API to get the page size for the versions we support (MacOS/swift).  When I thought about it all the platforms with bigger pages are multiples of 4K (i.e. 16K for arm M1) so this code will work.  It would just be a little less efficient because it breaks up the read in 4K chunks instead of 16K. It seems weird to make the syscall in a lldb extension but maybe that is just me.

Reply:
The LLDBServices::GetPageSize function in this file also and has always returned 4K.  I guess this one should be fixed for the arm platforms.

Reply:
Never mind all that. I decided to do what I did in createdump for the arm page size and use sysconf with a global var.

Reply:
The problem is sysconf will tell you the page size of where you are analyzing, and not necessarily of the target dump.

Reply:
I should have thought of that though I think people mostly use lldb on the architecture that the coredump was generated.  I'm not sure you can even load an arm64 linux dump on a x64 linux/lldb.

Reply:
PAGE SIZE on Linux is configuration, so it's regardless of the architecture. The 4k page may be slower, but it may be less likely to fault.

Reply:
I'm going to revert the sysconf changes.